### PR TITLE
Update code-generators and fix discovered bugs

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Update the code generators to 0.31.1 so that we can
consume new features, like options tracker
in fake clients.

This update exposed a bug in virtctl imageupload ds handling.
Also, since with the new fakeclient we are able to track the options
passed to the requests, the dryrun test in the pod-eviction-admitter
was re-enabled.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Update code-generators to 1.31.1
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
regarding the imageupload fix: FYI @codingben
/cc @0xFelix @xpivarc

**Comment by fossedihelm** [comment]
@acardace for the re-enabling of the pod-eviction-admitter test

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

**Review by xpivarc — COMMENTED** [review]
/approve
/hold

File: hack/builder/Dockerfile [hunk 1]
```diff
@@ -75,12 +75,12 @@ RUN set -x && \
     source /etc/profile.d/gimme.sh && \
     go install -v golang.org/x/tools/cmd/goimports@d5fe738 && \
     go install -v mvdan.cc/sh/v3/cmd/shfmt@v3.1.1 && \
-    go install -v k8s.io/code-generator/cmd/conversion-gen@v0.20.2  && \
-    go install -v k8s.io/code-generator/cmd/deepcopy-gen@v0.20.2 && \
-    go install -v k8s.io/code-generator/cmd/defaulter-gen@v0.20.2  && \
+    go install -v k8s.io/code-generator/cmd/conversion-gen@v0.31.1  && \
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply hack/builder/Dockerfile:78]: > what si means?
"SO" it's a typo :)

File: pkg/virtctl/imageupload/imageupload.go [hunk 2]
```diff
@@ -931,15 +931,14 @@ func (c *command) handleEventErrors(pvcName, dvName string) error {
 
 func (c *command) handleDataSource() error {
 	ds, err := c.client.CdiClient().CdiV1beta1().DataSources(c.namespace).Get(context.Background(), c.name, metav1.GetOptions{})
-	if err != nil && !k8serrors.IsNotFound(err) {
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virtctl/imageupload/imageupload.go:934]: Yes. I discovered that exactly because a unit test was failing after the client-gen update

**Comment by fossedihelm** [comment]
@0xFelix @xpivarc Fixed the typo in the commit message

File: hack/dockerized [hunk 3]
```diff
@@ -12,7 +12,7 @@ fi
 
 fail_if_cri_bin_missing
 
-kubevirt_builder_version="2410101324-d6011ee28e"
+kubevirt_builder_version="2410110922-18d7a3186d"
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply hack/dockerized:15]: Please remove hold @brianmcarey

[the remark this answers is not in the document]
Comment (brianmcarey) [reply hack/dockerized:15]: Yes I worked with @fossedihelm on getting this new builder image published - it should include all the recent changes to the builder from main  - lgtm 

/hold cancel

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**